dbms/src/Storages/IStorage.cpp

```diff
@@ -314,10 +314,10 @@ bool IStorage::isVirtualColumn(const String & column_name) const
     return getColumns().get(column_name).is_virtual;
 }
 
-TableStructureReadLockHolder IStorage::lockStructureForShare(const String & query_id)
+TableStructureReadLockHolder IStorage::lockStructureForShare()
 {
     TableStructureReadLockHolder result;
-    result.structure_lock = structure_lock->getLock(RWLockImpl::Read, query_id);
+    result.structure_shared_lock = std::make_shared<std::shared_lock<std::shared_mutex>>(structure_lock);
```

Comment:
Can we guarantee that the lock is always released by the same thread, that aquired it? Otherwise we are going to have UB!